src/main/java/com/javadiscord/javabot/commands/reaction_roles/ReactionRoles.java

```diff
@@ -1,152 +1,85 @@
 import net.dv8tion.jda.api.interactions.components.ButtonStyle;
 import net.dv8tion.jda.api.requests.restaction.interactions.ReplyAction;
-import org.bson.Document;
 
+import java.time.Instant;
 import java.util.ArrayList;
 import java.util.Date;
 import java.util.List;
 
-import static com.javadiscord.javabot.events.Startup.mongoClient;
-import static com.mongodb.client.model.Filters.eq;
-
-// TODO: Redo implementation of this to be much cleaner.
-@Deprecated(forRemoval = true)
 public class ReactionRoles implements SlashCommandHandler {
+
     @Override
     public ReplyAction handle(SlashCommandEvent event) {
-        return switch (event.getSubcommandName()) {
-            case "create" -> create(event,
-                    event.getOption("channel").getAsMessageChannel(),
-                    event.getOption("messageid").getAsString(),
-                    event.getOption("emote").getAsString(),
-                    event.getOption("button-label").getAsString(),
-                    event.getOption("role").getAsRole());
-            case "delete" -> delete(event,
-                    event.getOption("messageid").getAsString(),
-                    event.getOption("button-label").getAsString(),
-                    event.getOption("emote").getAsString());
-            default -> Responses.warning(event, "Unknown subcommand.");
-        };
+        try {
+            return switch (event.getSubcommandName()) {
+                case "create" -> createReactionRole(event);
+                case "delete" -> deleteReactionRole(event);
+                default -> Responses.warning(event, "Unknown subcommand.");
+            };
+        } catch (Exception e) { return Responses.error(event, "```" + e.getMessage() + "```"); }
     }
 
-    private ReplyAction create(SlashCommandEvent event, MessageChannel channel, String mID, String emote, String buttonLabel, Role role) {
-
-        MongoDatabase database = mongoClient.getDatabase("other");
-        MongoCollection<Document> collection = database.getCollection("reactionroles");
-
-        BasicDBObject criteria = new BasicDBObject()
-                .append("guild_id", event.getGuild().getId())
-                .append("channel_id", channel.getId())
-                .append("message_id", mID)
-                .append("button_label", buttonLabel)
-                .append("emote", emote);
-
-        if (collection.find(criteria).first() == null) {
-
-            Document doc = new Document()
-                    .append("guild_id", event.getGuild().getId())
-                    .append("channel_id", channel.getId())
-                    .append("message_id", mID)
-                    .append("role_id", role.getId())
-                    .append("button_label", buttonLabel)
-                    .append("emote", emote);
+    private String buttonId(Role role, boolean b) {
+        return "reaction-role:" + role.getId() + ":" + b + ":" + Instant.now();
+    }
```

Comment:
It would be nice if you could add documentation somewhere on this specific interaction id format.